docs/6.Contribution/Contribute New Terraform Provider.md

```diff
@@ -57,34 +61,37 @@ Add a placeholder file at `tests/terraform/checks/resource/linode/__init__.py`
 Create the folder `checkov/checkov/terraform/checks/resource/linode` and add `authorized_keys.py`:
 
 ```python
-from checkov.common.models.enums import CheckCategories, CheckResult
-from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceCheck
+from typing import Any
 
-class authorized_keys(BaseResourceCheck):
-    def __init__(self):
+from checkov.common.models.consts import ANY_VALUE
+from checkov.common.models.enums import CheckCategories
+from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceValueCheck
+
+class AuthorizedKeys(BaseResourceValueCheck):
+    def __init__(self) -> None:
         name = "Ensure SSH key set in authorized_keys"
         id = "CKV_LIN_2"
-        supported_resources = ['linode_instance']
-        categories = [CheckCategories.GENERAL_SECURITY]
+        supported_resources = ("linode_instance",)
+        categories = (CheckCategories.GENERAL_SECURITY,)
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
-    def scan_resource_conf(self, conf):
-        if 'authorized_keys' in conf:
-            if conf['authorized_keys']:
-                return CheckResult.PASSED
-        return CheckResult.FAILED
+    def get_inspected_key(self) -> str:
+        return "authorized_keys"
 
+    def get_expected_value(self) -> Any:
+        return ANY_VALUE
 
-check = authorized_keys() 
+
+check = AuthorizedKeys() 
 ```
 
 And also add `checkov/terraform/checks/resource/linode/__init__.py`:
 
 ```python
-from os.path import dirname, basename, isfile, join
-import glob
-modules = glob.glob(join(dirname(__file__), "*.py"))
-__all__ = [ basename(f)[:-3] for f in modules if isfile(f) and not f.endswith('__init__.py')]
+from pathlib import Path
```

Comment:
does this change break backward comptability?

Reply:
nope, no backwards compatibility issues. It does the same as before, but is the modern way to do it 🙂